internal/cmdrpkgupdate/discover.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmdrpkgupdate
+
+import (
+	"fmt"
+	"strings"
+
+	porchapi "github.com/GoogleContainerTools/kpt/porch/api/porch/v1alpha1"
+	configapi "github.com/GoogleContainerTools/kpt/porch/api/porchconfig/v1alpha1"
+	"github.com/spf13/cobra"
+	"golang.org/x/mod/semver"
+	"k8s.io/cli-runtime/pkg/printers"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (r *runner) discoverUpdates(cmd *cobra.Command, args []string) error {
+	var errs []string
```

Comment:
Is this actually used. I can't see where errors are added to the slice.

Reply:
Ah thanks for the catch, I messed this up while refactoring. Fixed to actually use it when finding package revisions.